### ISSUE TITLE
java.lang.RuntimeException: java.lang.RuntimeException: Table/View Not Found: <xxxx>

### ISSUE THREAD

**Issue by cpedrero** [Stratio/Crossdata#587]
Hi mates! we was trying to test in our project a query over Crossdata 1.3.0-RC3-SNAPSHOT and when Cassandra table is not defined in Crossdata Catalog yet, I get a RuntimeException like thisone:

```
Caused by: java.lang.Exception: java.lang.RuntimeException: Table/View Not Found: test_table
	at com.stratio.crossdata.server.actors.JobActor$$anonfun$com$stratio$crossdata$server$actors$JobActor$$receive$1.applyOrElse(JobActor.scala:153)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:467)
	at com.stratio.crossdata.server.actors.JobActor.aroundReceive(JobActor.scala:96)
```

My code is similar to your driver example over Cassandra:

```
import akka.actor.{Actor, ActorLogging, Props}
import akka.http.scaladsl.model.StatusCodes._
import com.datio.daas.core.bean.XDBeanQuery
import com.datio.daas.core.crossdata.SqlQueryCreator
import com.datiobd.daas.model.bean.BeanErrorRequest
import com.stratio.crossdata.driver.Driver
import com.stratio.crossdata.driver.config.DriverConf
import scala.collection.JavaConverters._
class XDActor(targets: List[String]) extends Actor with ActorLogging {

  val conf = new DriverConf()
    .setFlattenTables(false)
    .setTunnelTimeout(30)
    .setClusterContactPoint(targets.asJava)

  lazy val driver = Option(Driver.getOrCreate(conf))
  override def receive: Receive = {
    case xdBean: XDBeanQuery => {
      try {
        for {
          xdDriver <- driver
        } {
          sender !
            xdDriver.sql(SqlQueryCreator(xdBean.sqlQuery).createSelect).waitForResult().resultSet
        }
      } catch {
        case ex: Exception =>
          sender !  Left(BeanErrorRequest(BadRequest.intValue, ex.getCause.getMessage))
      }finally {
        driver.foreach(_.stop())
      }
    }

  }
}
```

Maybe I did something wrong or maybe we should get an ErrorSQLResult o something like that to manage the error message.
Could you help us about that? Thanks!

**Comment by miguel0afd** [comment]
Hi @cpedrero,

the problem is related to how the SQLResult is managed. The SQLResult is the object returned by the method _waitForResult()_. Therefore, this SQLResult can be an ErrorSQLResult (which doesn't contain a resultSet) or a SuccessfulSQLResult (which does contain a resultSet). A nice way to deal with this situation would be something like:

xdDriver.sql(SqlQueryCreator(xdBean.sqlQuery).createSelect).waitForResult() **match {
        case error: ErrorSQLResult => sys.error(error.message)
        case sqlResult: SuccessfulSQLResult => println(s"Result length: ${sqlResult.resultSet.length}")
}**

By the way, the Crossdata version 1.3.0 has been already released.

I hope it helps.

**Comment by cpedrero** [comment]
Thanks for your answer Miguel Ángel.
I was worng with my implementation and your code is a good aproach for my goal.
But I have changed my code and in the error field of ErrorSQLResult I get:
`java.lang.RuntimeException: Table/View Not Found: test_table`
instead "Table/View Not Found: test_table"

**Comment by miguel0afd** [comment]
This issue has been fixed in the branch-1.4. We'll be releasing the version 1.4.0 shortly
